src/Build/Logging/BinaryLogger/Postprocessing/CleanupScope.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Build.Logging;
+
+internal readonly struct CleanupScope : IDisposable
```

Comment:
NIT: This generic usage class could be in Framework assembly for reuse.

Reply:
There are other candidates. Right now I want to keep the change scoped. Once it makes in, I'd want to move the usefull helpers out